core/src/main/java/software/amazon/awssdk/core/AwsRequestOverrideConfig.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core;
+
+import java.util.Optional;
+
+import software.amazon.awssdk.core.auth.AwsCredentialsProvider;
+
+/**
+ * Request-specific configuration overrides for AWS service clients.
+ */
+public final class AwsRequestOverrideConfig extends SdkRequestOverrideConfig {
+    private final AwsCredentialsProvider awsCredentialsProvider;
+
+    private AwsRequestOverrideConfig(Builder builder) {
+        super(builder);
+        this.awsCredentialsProvider = builder.awsCredentialsProvider();
+    }
+
+    public Optional<AwsCredentialsProvider> awsCredentialsProvider() {
+        return Optional.ofNullable(awsCredentialsProvider);
+    }
+
+    @Override
+    public Builder toBuilder() {
+        return new BuilderImpl(this);
+    }
+
+    public static Builder builder() {
+        return new BuilderImpl();
+    }
+
+    public interface Builder extends SdkRequestOverrideConfig.Builder<Builder> {
+        Builder awsCredentialsProvider(AwsCredentialsProvider awsCredentialsProvider);
+
+        AwsCredentialsProvider awsCredentialsProvider();
```

Comment:
We'll want comments on all of this. I assume that's pending review, but I wanted to say it anyway.

Reply:
Done